Dockerfile

```diff
@@ -2,73 +2,80 @@ FROM phusion/baseimage:0.9.18
 
 MAINTAINER Luke Campbell <luke.campbell@rpsgroup.com>
 
-RUN apt-get update && apt-get install -y \
-      git \
-      libffi-dev \
-      libhdf5-dev \
-      libnetcdf-dev \
-      netcdf-bin \
-      libssl-dev \
-      libxml2-dev \
-      libxslt1-dev \
-      nodejs \
-      npm \
-      python-dev \
-      python-pip \
-      libudunits2-0 \
-      redis-tools \
-      libgeos-dev \
-      libpng12-dev \
-      libfreetype6-dev \
-      && rm -rf /var/lib/apt/lists/*
-
-COPY requirements.txt /requirements.txt
-RUN pip install -r /requirements.txt
-# The debian version of six caused a bug in Flask
-RUN pip install -U 'numpy==1.11.0' 'netCDF4==1.2.4'
-RUN pip install -U six
-RUN pip install cc-plugin-ncei
-RUN pip install gunicorn supervisor
-
-# install node dependencies and rename link node binary
-RUN npm install -g bower grunt-cli && npm cache clear && \
-    ln -s /usr/bin/nodejs /usr/bin/node
-
-# Directory for the project
-RUN mkdir /usr/lib/cchecker-web
-
-# Copy over project contents
-COPY cchecker_web /usr/lib/cchecker-web/cchecker_web
-COPY .bowerrc app.py Gruntfile.js Assets.json bower.json package.json setup.py worker.py /usr/lib/cchecker-web/
-COPY contrib/config/config.yml /usr/lib/cchecker-web/config.yml
-
-RUN useradd -m web
-RUN chown -R web:web /usr/lib/cchecker-web
-
-# Install project dependencies
-WORKDIR /usr/lib/cchecker-web
-USER web
-
-RUN npm install
-RUN bower install
-RUN grunt
-
+# General dependencies:
+RUN apt-get update && \
+    apt-get install -y wget git build-essential && \
+    rm -rf /var/lib/apt/lists/*
+
+# Install nodejs/npm and friends:
+RUN (curl -sL https://deb.nodesource.com/setup_6.x | bash) && \
+    apt-get update && \
+    apt-get install -y nodejs && \
+    npm install -g grunt-cli && \
+    rm -rf /var/lib/apt/lists/*
+
+# Install conda/python
+RUN echo 'export PATH=/opt/conda/bin:$PATH' > /etc/profile.d/conda.sh && \
+    wget --quiet https://repo.continuum.io/miniconda/Miniconda3-4.0.5-Linux-x86_64.sh -O ~/miniconda.sh && \
+    /bin/bash ~/miniconda.sh -b -p /opt/conda && \
+    rm ~/miniconda.sh && \
+    export PATH="/opt/conda/bin:${PATH}" && \
+    conda config --set always_yes yes --set changeps1 no && \
+    conda config --set show_channel_urls True && \
+    conda config --add create_default_packages pip && \
+    echo 'conda 4.0.*' >> /opt/conda/conda-meta/pinned && \
+    conda update conda && \
+    conda config --add channels conda-forge && \
+    conda clean --all --yes
+ENV PATH=/opt/conda/bin:$PATH
+
+# Install container dependencies:
+    # redis: Required for redis-cli ping in 50_configure
+    # gunicorn: Required for webui
+conda install redis=3.2.0 gunicorn=19.6.0 && \
```

Comment:
That's most of the point of using conda in the first place.  Instead of monkeying around with system level packages to get netcdf and scipy installed/working, conda handles it.  Who wants to remember to install `libblas1-dev`.